nibabel/testing/__init__.py

```diff
@@ -57,6 +59,21 @@ def assert_allclose_safely(a, b, match_nans=True):
     assert_true(np.allclose(a, b))
 
 
+def isiterable(iterable):
+    try:
+        for _ in iterable:
+            pass
+    except:
+        return False
+
+    return True
```

Comment:
You are right. However, `isinstance(val, collections.Iterable)` simply checks if the object has an `__iter__` method instead of testing that you can actually iterate through it. The same goes for testing that calling `iter` is not failling.

I will agree that `isiterable` might not be the right name for this function but I lack a better one right now (if you have suggestion let me know). I use this `isiterable` function to loop through all elements of a given object, catching exception if any (could happen for my objects).